cclient/types/types.c

```diff
@@ -108,3 +112,86 @@ int_array* string_to_int_array(char* in) {
   }
   return in_arr;
 }
+
+retcode_t flex_hash_to_trytes(flex_hash_t* hash, char* trytes) {
+  size_t trits_len = 0;
+  if (trytes == NULL) {
+    return RC_CCLIENT_FLEX_TRITS;
+  }
+  trits_len = flex_trits_to_trytes((signed char*)trytes, hash->len_trits,
+                                   hash->trits, hash->len_flex, hash->len_flex);
+  if (trits_len == 0) {
+    return RC_CCLIENT_FLEX_TRITS;
+  }
+  return RC_OK;
+}
+
+retcode_t trytes_to_flex_hash(flex_hash_t* hash, const char* trytes) {
+  size_t str_len = strlen(trytes);
+  size_t flex_len = str_len * 3;
+  flex_trit_t* flex_trytes =
+      (flex_trit_t*)malloc(sizeof(flex_trit_t) * flex_len);
+  hash->len_trits = flex_trits_from_trytes(
+      flex_trytes, flex_len, (const tryte_t*)trytes, str_len, str_len);
+
+  if (hash->len_trits > 0) {
+    hash->trits = flex_trytes;
+    hash->len_flex = flex_len;
+  } else {
+    return RC_CCLIENT_FLEX_TRITS;
+  }
+  return RC_OK;
+}
+
+flex_hash_array_t* flex_hash_array_new() { return NULL; }
+
+flex_hash_array_t* flex_hash_array_append(flex_hash_array_t* head,
+                                          const char* trytes) {
+  flex_hash_array_t* elt =
+      (flex_hash_array_t*)malloc(sizeof(flex_hash_array_t));
```

Comment:
OOM check, this will affect commands now since they will now have to check for `flex_hash_array_append` returned value